micrometer-core/src/main/java/io/micrometer/core/instrument/composite/AbstractCompositeMeter.java

```diff
@@ -49,9 +49,11 @@ final Iterable<T> getChildren() {
     }
 
     T firstChild() {
-        final Iterator<T> i = children.values().iterator();
-        if (i.hasNext())
-            return i.next();
+        for (T next : children.values()) {
```

Comment:
This _should not_ be on the "hot path" (i.e.: recording a measurement or retrieving meters) but it will be called every time a value is fetched for publishing.

I'm thinking if there might be another way to implement this: modifying the `children` collection so that it has some priority ordering so the non-noop meters in it take precedence and will be "in front" of the collection (if any). That solution though might be more problematic since it seems a bit more complicated and it can be on a "hotter" path (meter creation).

Reply:
Have you seen [this PR](https://github.com/micrometer-metrics/micrometer/pull/1443/commits/fa1c14e5b3cdf1139df7cfa39a38ee8e025dce28)? The approach is completely different but certainly more efficient. We should understand if there is any hidden side effect if the meter is not added..

Reply:
I guess https://github.com/micrometer-metrics/micrometer/pull/1442 is also connected.
I would not do what is in https://github.com/micrometer-metrics/micrometer/pull/1443, it is somewhat similar to the priority ordering solution I explained before so its path is "hotter" and it omits meters from the registry.